utils/src/main/java/org/corfudb/utils/lock/Lock.java

```diff
@@ -105,34 +99,53 @@
     @Getter
     private Map<LockStateType, LockState> states = new HashMap<>();
 
+
+    private final LockConfig lockConfig;
     /**
      * Constructor
      *
      * @param lockId
      * @param lockListener
      * @param clientContext
      */
-    public Lock(LockDataTypes.LockId lockId, LockListener lockListener, LockClient.ClientContext clientContext) {
+    public Lock(LockDataTypes.LockId lockId, LockListener lockListener, LockClient.ClientContext clientContext,
+                LockConfig lockConfig) {
         this.lockId = lockId;
         this.clientContext = clientContext;
         this.lockListener = lockListener;
         this.eventConsumer = Executors.newSingleThreadExecutor(new
                 ThreadFactoryBuilder().setNameFormat("lock-event-consumer").build());
-
+        this.lockConfig = lockConfig;
```

Comment:
To avoid code repetition this could just call the other constructor with NO_LEASE as init state. 
this(lockId, lockListener, clientContext, lockConfig, LockStateType.NO_LEASE)

Reply:
ok